examples/model_selection/plot_grid_search_refit_callable.py

```diff
@@ -0,0 +1,113 @@
+"""
+==================================================
+Balance model complexity and cross-validated score
+==================================================
+
+This example balances model complexity and cross-validated score by
+finding a decent accuracy within 1 standard deviation of the best accuracy
+score while minimising the number of PCA components.
+
+The figure shows the trade-off between cross-validated score and number of
+PCA components. The balanced case is when n_components=6 and accuracy=0.80,
+which falls into the range within 1 standard deviation of the best accuracy
+score.
+"""
+# Author: Wenhao Zhang <wenhaoz@ucla.edu>
+
+print(__doc__)
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+from sklearn.datasets import load_digits
+from sklearn.decomposition import PCA
+from sklearn.model_selection import GridSearchCV
+from sklearn.pipeline import Pipeline
+from sklearn.svm import LinearSVC
+
+
+def lower_bound(cv_results):
+    """
+    Calculate the lower bound within 1 standard deviation
+    of the best `mean_test_scores`.
+
+    Parameters
+    ----------
+    cv_results : dict of numpy(masked) ndarrays
+        See attribute cv_results_ of `GridSearchCV`
+
+    Returns
+    -------
+    float
+        Lower bound within 1 standard deviation of the
+        best `mean_test_score`.
+    """
+    best_score_idx = np.argmax(cv_results['mean_test_score'])
+
+    return (cv_results['mean_test_score'][best_score_idx]
+            - cv_results['std_test_score'][best_score_idx])
+
+
+def refit_callable(cv_results):
+    """
+    Balance model complexity with cross-validated score.
+
+    Parameters
+    ----------
+    cv_results : dict of numpy(masked) ndarrays
+        See attribute cv_results_ of `GridSearchCV`.
+
+    Return
+    ------
+    int
+        Index of a model that has the fewest PCA components
+        while has test score within 1 standard deviation of the best
+        `mean_test_score`.
+    """
+    threshold = lower_bound(cv_results)
+    candidate_idx = np.flatnonzero(cv_results['mean_test_score'] >= threshold)
+    best_idx = candidate_idx[cv_results['param_reduce_dim__n_components']
+                             [candidate_idx].argmin()]
+    return best_idx
+
+
+pipe = Pipeline([
+        ('reduce_dim', PCA(random_state=42)),
+        ('classify', LinearSVC(random_state=42)),
+])
+
+N_FEATURES_OPTIONS = [2, 4, 6, 8]
```

Comment:
I don't think you're using this variable later, you can just remove it and put the values in the dict definition bellow.